ncp.c

```diff
@@ -1556,7 +1558,7 @@ cancel_conn_threads(struct conn *conn)
   if (pthread_equal(conn->conn_to_net_thread, pthread_self()) ||
       pthread_equal(conn->conn_from_sock_thread, pthread_self())) {
     // conn_to_sock is the main thread, which waits for the others to join - they just exit.
-    if (ncp_debug) printf("%%%% Conn %p thread %p (%s) terminating\n", conn, pthread_self(), conn_thread_name(conn));
+    if (ncp_debug) printf("%%%% Conn %p thread %p (%s) terminating\n", conn, (void *)pthread_self(), conn_thread_name(conn));
```

Comment:
I'm not sure what the best way to printf a pthread_t is.

Reply:
This is as good as any, it's just to get the unique identity of the thread.